### ISSUE TITLE
gen_nccl_ops.nccl_all_reduce will influence optimizer!!!

### ISSUE THREAD

**Issue by zihao-lu** [tensorflow/tensorflow#42558]
My version is tf 1.13, I use multi-GPUs to  train a model, when i minimize the loss in 0-gpu, 1-gpu's is not minimized, there is someting strange.
so this is my code 

<img width="701" alt="20200821-224654(WeLinkPC)" src="https://user-images.githubusercontent.com/34296901/90903315-58d97280-e400-11ea-9f58-a2a56ed87b76.png">

then when i run the code , it will keep stopping at a point. After debuging, i find that when  i use nccl_all_reduce in syncBN, this will appear, like this

<img width="588" alt="20200821-224658(WeLinkPC)" src="https://user-images.githubusercontent.com/34296901/90903324-5e36bd00-e400-11ea-87a9-bb44d21d66cd.png">

when i comment the two line, everything is ok.
By th way, i minimize all losses in all gpus using compute_gradient and apply_gradient, when using nccl_reduce_all,  the model could't converge, after commentting it everything  is ok.

I think it's a bug.

**Comment by rmothukuru** [comment]
@zihao-lu,
Can you please paste the complete code (rather than sharing the screenshot) so that it would be easy for us to reproduce your issue. Thanks!

**Comment by rmothukuru** [comment]
@zihao-lu,
Can you please respond to the comment. Thanks!

**Comment by rmothukuru** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
